sessionctx/variable/sysvar.go

```diff
@@ -1958,6 +1958,23 @@ var defaultSysVars = []*SysVar{
 		s.OptPrefixIndexSingleScan = TiDBOptOn(val)
 		return nil
 	}},
+	{Scope: ScopeGlobal, Name: TiDBExternalTS, Value: strconv.FormatInt(DefTiDBExternalTS, 10), SetGlobal: func(ctx context.Context, s *SessionVars, val string) error {
```

Comment:
`SetGlobal` should only be called when the session executes `set @@global.tidb_external_ts` explicitly. I'm sure whether this assumption is true. Please make sure it will not be called by other cases like `rebuildSysVarCache`.

Reply:
The `TiDB_EXTERNAL_TS` has been added to `SkipSysvarCache` function, so it will be skipped in `rebuildSysVarCache` :thinking:. I've checked all references of `SetGlobal`, and it should be fine.